cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureSetup.java

```diff
@@ -88,7 +88,7 @@ private void throwExceptionWithDetails(AuthenticatedContext ac, CloudStack stack
             String message = String.format("%s image copy failed. You may try to execute the copy manually, in that case please copy %s to the 'images' " +
                             "container of the storage account '%s' in resource group '%s'. Reason of failure: %s",
                     image.getImageName(), imageCopyDetails.getImageSource(), imageCopyDetails.getImageStorageName(),
-                    imageCopyDetails.getImageResourceGroupName(), ExceptionUtils.getRootCause(ex).getMessage());
+                    imageCopyDetails.getImageResourceGroupName(), ex.getCause().getMessage());
```

Comment:
Out of curiosity, what is the reason behind this change? Is it intentional to do this here but not in the `else` branch below (in line 95)?

Reply:
In the case I have tested, the original Exception was thrown from `AzureStorageAccountBuilderService:47`, that had a cause from the Azure sdk, that had a cause of `com.fasterxml.jackson.databind.exc.MismatchedInputException`. So the root cause was not really useful for us, and we are parsing it in `AzureUtils.convertToCloudConnectorException` either way, so it is better to use that. But yeah, maybe it should be modified in the else branch too.

Reply:
The strange thing is, in prod logs we have lines like this:

> https://cldreastus.blob.core.windows.net/images/freeipa-cdh--1641901019.vhd image copy failed. You may try to execute the copy manually, in that case please copy https://cldreastus.blob.core.windows.net/images/freeipa-cdh--1641901019.vhd to the 'images' container of the storage account 'cbimgeu8301085c89d027a' in resource group 'cdppoc'. Reason of failure: Could not create storage account cbimgeu8301085c89d027a in resource group cdppoc

But when I try to reproduce it locally (even on 2.54 branch), I get errors like this:

> https://cldrwestus2.blob.core.windows.net/images/freeipa-cdh--1646402436.vhd image copy failed. You may try to execute the copy manually, in that case please copy https://cldrwestus2.blob.core.windows.net/images/freeipa-cdh--1646402436.vhd to the 'images' container of the storage account 'cbimgwu29d62091481040a03' in resource group 'dbajzath-azure-restricted'. Reason of failure: OnError while emitting onNext value: retrofit2.Response.class'

Reply:
So I figured it out, the azure sdk was not able to deserialize the customer's policy violation, but it can deserialize mine to construct a CloudException from it.
It became an IOException, which is caught here and transformed to RuntimeException: https://github.com/hortonworks/cloudbreak/blob/f33cf3843a1862082156e4335aa01960b0252c78/cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/client/AzureClient.java#L202
So in the caller mine and their exception goes to a different catch:
https://github.com/hortonworks/cloudbreak/blob/a5cd4405f48a8df4f2eb308233afaa51f488b1d4/cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/connector/resource/AzureStorageAccountBuilderService.java#L48
 So even with my changes, the error message will not contain the policy name, so I will set this PR back to draft.

Reply:
Oh, interesting finding indeed... Please let me know once the PR is ready for a re-review then. :smile: